bosond.cpp

```diff
@@ -77,13 +65,69 @@ void processArgs(int argc, char **argv) {
     }
 }
 
+// Initalise the command control interface (CCI) using SDK.
+void initCCI() {
+    libusb_device_handle* dev_h;
+
+    if (libusb_init(NULL) < 0) {
+        perror("Error: Failed libusb init");
+        exit(1);
+    }
+
+    if ((dev_h = libusb_open_device_with_vid_pid(NULL, 0x09cb, 0x4007)) == NULL) {
+        perror("couldn't open USB device");
+        exit(1);
+    }
+
+    Initialize(dev_h);
+}
+
+int logCameraInfo() {
+    uint32_t major, minor, patch;
+    if (int result = bosonGetSoftwareRev(&major, &minor, &patch)) {
+        std::cout << "failed to get software rev: " << result << '\n';
```

Comment:
Any reason therse aren't perror and exit right after? as is done elsewhere
same goes for below.

Otherwise should thay be +1 to indicate error?

Reply:
perror only works for retrieving the error for the last failed system call. There won't necessary be an error like this set for calls to the Boson SDK.

Returning negative numbers for error codes seems to be the idiomatic approach in C.